docs/scripting-element-object.rst

```diff
@@ -0,0 +1,744 @@
+
+Mouse events are not propagated immediately, to see consequences of click
+reflected in page source you must call :ref:`splash-wait`
+
+Example 1: get width and height of the element, calculate its center and click on it
+
+.. code-block:: lua
+
+    function main(splash)
+        -- ...
+        local element = splash:select('.element')
+        local bounds = element:bounds()
+        assert(element:mouse_click{x=bounds.width/2, y=bounds.height/2})
+        -- ...
+    end
+
+
+Example 2: click on the area above the element by 10 pixels
+
+.. code-block:: lua
+
+    function main(splash)
+        -- ...
+        local element = splash:select('.element')
+        assert(element:mouse_click{x=-10})
+        -- ...
+    end
+
+See more about mouse events in :ref:`splash-mouse-click`.
+
+
+.. _splash-element-mouse-hover:
+
+element:mouse_hover
+-------------------
+
+Trigger mouse hover (JavaScript mouseover) event on the element.
+
+**Signature:** ``ok, reason = element:mouse_hover{x=0, y=0}``
+
+**Parameters:**
+
+* x - optional, x coordinate relative to the left corner of the element
+* y - optional, y coordinate relative to the top corner of the element
+
+**Returns:** ``ok, reason`` pair. If ``ok`` is nil then error happened during the
+function call; ``reason`` provides an information about error type.
+
+**Async:** no.
+
+If x or y coordinate is not provided they will be set to 0 and the hover will be triggered
+on the left-top corner of the element. The coordinates can have a negative value which means
+the hover will be triggered outside of the element.
+
+Mouse events are not propagated immediately, to see consequences of hover
+reflected in page source you must call :ref:`splash-wait`
+
+Example 1: get width and height of the element, calculate its center and hover over it
+
+.. code-block:: lua
+
+    function main(splash)
+        -- ...
+        local element = splash:select('.element')
+        local bounds = element:bounds()
+        assert(element:mouse_hover{x=bounds.width/2, y=bounds.height/2})
+        -- ...
+    end
+
+
+Example 2: hover over the area above the element by 10 pixels
+
+.. code-block:: lua
+
+    function main(splash)
+        -- ...
+        local element = splash:select('.element')
+        assert(element:mouse_hover{x=-10})
+        -- ...
+    end
+
+See more about mouse events in :ref:`splash-mouse-hover`.
+
+
+.. _splash-element-styles:
+
+element:styles
+--------------
+
+Return the computed styles of the element.
+
+**Signature:** ``styles = element:styles()``
+
+**Returns:** ``styles`` is a table with computed styles of the element.
+
+**Async:** no.
+
+This method returns the result of JavaScript `Window.getComputedStyle()`_ applied on the element.
+
+Example: get all computed styles and return the ``font-size`` property.
+
+.. code-block:: lua
+
+    function main(splash)
+        -- ...
+        local element = splash:select('.element')
+        return element:styles()['font-size']
```

Comment:
note for the future: probably `element:style('font-size')` makes more sense - this is usually what developer wants, it is less code and less JS->Python->Lua conversions